### DISCUSSION TITLE
Migration from caip-72 to caip-122

### DISCUSSION THREAD

**Discussion by darrenkel** [WalletConnect/walletconnect-monorepo#3865] [Developer Support]
Would it be possible to migrate the auth requests from CAIP-72 to CAIP-122 ? Or support both?

122 allows us to specify the blokchain, so 
```
${domain} wants you to sign in with your **blockchain** account:
```

**Comment by bumblefudge** [comment]
Hey sorry, I'm not sure I understand the question.  CAIP-122 is meant to describe/specify the creation of a sign-in flow that generates an authN receipt conformant to CAIP-74.  There's not much reason to do CAIP-74 unless it's as an eventlog/replayable receipt of CAIP-122 events.  Is this a question about moving from a non-conformant, DIY authN signing flow to a conformant one?  Or a message about upgrading from CAIP-122 to CAIP-222?

**Comment by bumblefudge** [comment]
Also, I'm not super familiar with the specifics of what WC SDKs/tooling you might be using to facilitate this, so if I say anything that contradicts WC personnel, definitely assume I'm wrong and listen to them instead 😅

**Comment by darrenkel** [comment]
@bumblefudge maybe I've misunderstood, but my goal is to show the user in my wallet `Sign in with **blokchain**` rather than `Sign in with Ethereum`. I understood 122 to be an extension of 74.

From what I can see, WC only supports `Sign in with Ethereum`

https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-122.md#motivation

<img width="1213" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/107671032/940872af-d485-4145-a90d-c1814ea27ea1">

**Reply by bumblefudge** [reply]
oh, I see, you're good. caip-74 was written as a receipt for SIWE messages, and CAIP 122 defines a general template for SIWx messages that also produces CAIP74 receipts (a profile needs to be defined per-chain, tho). the WC relays recognize and know how to route CAIP122 objects if a VeChain dapp presents a VeChain wallet (and/or a Solana wallet and/or an EVM wallet and/or any other wallet that has a CAIP122 profile defined at namespaces.chainagnostic.org ). but it sounds like you're asking if the WC SDKs already include dapp code for presenting that message to the wallets... there i think the answer is still no but WC folks can chime in if they have guidance on how to get VeChain dapps on the CAIP-122 train
